app/views/users/show.html.erb

```diff
@@ -4,18 +4,24 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
```

Comment:
You could keep the logic here. One workaround is to create your own method in policy such that it is dissociated from `show?`. For example, in your User Policy, you could create a `show_photos?` policy with this boolean expression, and here call `if policy(@user).show_photos?`